src/gluonts/model/rotbaum/_model.py

```diff
@@ -124,29 +125,50 @@ def _create_xgboost_model(model_params: Optional[dict] = None):
             }
         return xgboost.sklearn.XGBModel(**model_params)
 
-    def fit(self, x_train, y_train, max_sample_size=None, seed=1, **kwargs):
+    def fit(
+        self,
+        x_train,
+        y_train,
+        max_sample_size=None,
+        seed=1,
+        x_train_is_dataframe=False,
+        **kwargs
+    ):
         """
         Fits self.model and partitions R^n into cells. More accurately,
         it creates a dictionary taking predictions on train to lists of
         associated true values, and puts it in self.cell_values.
 
         Parameters
         ----------
-        x_train: list
-            list of lists
-        y_train: list
+        x_train: pd.DataFrame (if x_train_is_dataframe) or list (else)
```

Comment:
Maybe move type descriptions as type annotations into the arguments?

Reply:
Done.